xapian-letor/ranker/ranker.cc

```diff
@@ -351,19 +351,35 @@ Ranker::labelcomparer(const FeatureVector & firstfv, const FeatureVector& second
     return firstfv.get_label() > secondfv.get_label();
 }
 
-std::vector<Xapian::docid>
-Ranker::rank(const Xapian::MSet & mset, const string & model_key, const Xapian::FeatureList & flist)
+class ScoreIterator {
+    std::vector<FeatureVector>::const_iterator it;
+
+  public:
+    explicit ScoreIterator(const std::vector<FeatureVector>::const_iterator it_) : it(it_) { }
+
+    double operator*() const { return it->get_score(); }
+
+    void operator++() { ++it; }
+
+    bool operator!=(const ScoreIterator& o) { return it != o.it; }
+
+    bool operator==(const ScoreIterator& o) { return it == o.it; }
+
+    Xapian::doccount operator-(const ScoreIterator& o) { return Xapian::doccount(it - o.it); }
+};
+
+void
+Ranker::rank(Xapian::MSet & mset, const string & model_key, const Xapian::FeatureList & flist)
 {
+    if (mset.empty()) {
+	return;
+    }
     LOGCALL(API, std::vector<Xapian::docid>, "Ranker::rank", mset | model_key | flist);
```

Comment:
`LOGCALL` should be the first thing in the function (didn't I already say that?)

Also, you've change the return type, but not updates the `LOGCALL` to match.